### PR TITLE
Update embedded JS/TS analyzer to 10.1.0.21143

### LINKED ISSUE #3929: [Infra] Update build-time JS/TS zipped artefact name
### Description

BLOCKED - requires the release of SonarJS/TS v10.1

The name of the JS/TS zipped artefact embedded in the jar has changed: `eslint-bridge-1.0.0.tgz is now sonarjs-1.0.0.tgz`.

TODO
- [x] Update build-time processing
- [x] Check that names the artefacts in the zip have not changed

### PR DESCRIPTION
**Pull request by duncanp-sonar**
Fixes #3929 and #3972

### CODE REVIEW

File: build/DownloadSonarPlugin/DownloadAndExtractSonarJS.cs [hunk 1]
```diff
@@ -103,7 +103,7 @@ public override bool Execute()
             var fileList = FindFiles(perVersionPluginFolder);
             FilesToEmbed = fileList.ToArray();
 
-            PackageDirectoryToEmbed =   Path.Combine(perVersionPluginFolder, SourceRelativePackageDirectory);
```
Comment (duncanp-sonar) [inline build/DownloadSonarPlugin/DownloadAndExtractSonarJS.cs:106]: Miscellaneous: removed whitespace

File: build/DownloadSonarPlugin/DownloadAndExtractSonarJS.cs [hunk 2]
```diff
@@ -51,10 +51,10 @@ public class DownloadAndExtractSonarJS: Task
         //          package\package.json
 
         // The archive containing the eslintbridge
-        private const string SourceEsLintBridgeFilePattern = "eslint-bridge-*.tgz";
+        private const string SourceEsLintBridgeFilePattern = "sonarjs-*.tgz";
```
Comment (duncanp-sonar) [inline build/DownloadSonarPlugin/DownloadAndExtractSonarJS.cs:54]: This related to #3929.
At build time, we download the SonarJS jar, unzip it, and extract some files.
The name of one of the files we extract has changed, so this string needed to be updated.

File: build/DownloadSonarPlugin/DownloadAndExtractSonarJS.cs [hunk 3]
```diff
@@ -51,10 +51,10 @@ public class DownloadAndExtractSonarJS: Task
...
 
         // Sub-folder into which the tar file should be unzipped
-        public const string TargetEslintBridgeFolderName = "eslint-bridge";
+        public const string TargetEslintBridgeFolderName = "sonarjs";
```
Comment (duncanp-sonar) [inline build/DownloadSonarPlugin/DownloadAndExtractSonarJS.cs:57]: This is the name of a temporary working folder used when extracting files from the jar.
I didn't really need to rename it, but it made it easy to be sure that we were definitely only getting files from the embedded file (and also easy to compare against the previous version as a sanity check of the number/names of files unzipped).

File: src/EmbeddedSonarAnalyzer.props [hunk 4]
```diff
@@ -6,7 +6,7 @@
     <!-- Note: Guide on how to update the analyzers is on the xtranet! -->
     <EmbeddedSonarAnalyzerVersion>8.55.0.65544</EmbeddedSonarAnalyzerVersion>
     <EmbeddedSonarCFamilyAnalyzerVersion>6.43.0.61486</EmbeddedSonarCFamilyAnalyzerVersion>
-    <EmbeddedSonarJSAnalyzerVersion>10.0.1.20755</EmbeddedSonarJSAnalyzerVersion>
+    <EmbeddedSonarJSAnalyzerVersion>10.1.0.21143</EmbeddedSonarJSAnalyzerVersion>
```
Comment (duncanp-sonar) [inline src/EmbeddedSonarAnalyzer.props:9]: #3972 - update to the new version of the analyzer.

To test the change, I added JS/TS files to the sample project that test a new rule that was added in the new analyzer. Issues are raised, so it looks like the correct files are being extracted from the jar and embedded in the VSIX.

image

**Review by bigfluffycookie — APPROVED** [review]
LGTM

**Pull request merged**